src/databases/Mili/avtMiliFileFormat.C

```diff
@@ -74,4384 +53,3837 @@ extern "C" {
-        // cells.
-        //
-        connectivity[dom][mesh_id] = vtkUnstructuredGrid::New();
 
-        const int n_elem_types = 8;
-        static int elem_sclasses[n_elem_types] =
-        {
-            M_TRUSS, M_BEAM, M_TRI, M_QUAD, M_TET, M_PYRAMID, M_WEDGE, M_HEX
-        };
-        static int conn_count[n_elem_types] =
-        {
-            2, 3, 3, 4, 4, 5, 6, 8
-        };
-        
-        //
-        // Make one pass through the data and read all of the connectivity
-        // information.
-        //
-        vector < vector<int *> > conn_list;
-        vector < vector<int *> > mat_list;
-        vector < vector<int> > list_size;
-        conn_list.resize(n_elem_types);
-        list_size.resize(n_elem_types);
-        mat_list.resize(n_elem_types);
-
-        max_zone_label_lengths[dom] = 0;
-
-        int i, j, k;
-        ncells[dom][mesh_id] = 0;
-        int ncoords = 0;
+                numCellsPerClass.push_back(nCells);
+                classNames.push_back(string(shortName));
+                nDomCells += nCells;
 
-        int total_elems = 0;
-
-        for (i = 0 ; i < n_elem_types ; i++)
-        {
-            int args[2];
-            args[0] = mesh_id;
-            args[1] = elem_sclasses[i];
-            int ngroups = 0;
-            mc_query_family(dbid[dom], QTY_CLASS_IN_SCLASS, (void*) args, NULL,
-                            (void*) &ngroups);
-            for (j = 0 ; j < ngroups ; j++)
-            {
-                int nelems;
-                mc_get_class_info(dbid[dom], mesh_id, elem_sclasses[i], j,
-                                  short_name, long_name, &nelems);
-                 
-                int *conn = new int[nelems * conn_count[i]];
-                int *mat = new int[nelems];
-                int *part = new int[nelems];
-                mc_load_conns(dbid[dom], mesh_id, short_name, conn, mat, part);
-
-                conn_list[i].push_back(conn);
-                mat_list[i].push_back(mat);
-                list_size[i].push_back(nelems);
-                connectivity_offset[dom].push_back(ncells[dom][mesh_id]);
-                element_group_name[dom].push_back(short_name);
-                group_mesh_associations[dom].push_back(mesh_id);
-                ncells[dom][mesh_id] += list_size[i][j];
-
-                ncoords += list_size[i][j] * (conn_count[i] + 1);
-                delete [] part;
-
-                PopulateZoneLabels(dbid[dom], mesh_id, short_name, dom, 
-                                   total_elems, nelems);
+                miliMetaData[meshId]->GetClassMDByShortName(shortName)->
+                    SetConnectivityOffset(dom, offset);
+                miliMetaData[meshId]->GetClassMDByShortName(shortName)->
+                    SetNumElements(dom, nCells);
+                offset += nCells;
             }
         }
 
+        // TODO: should we set the num elements for classes not set
```

Comment:
Did you satisfy this todo?

Reply:
Just resolved this. I had been lazy and avoided implementing doing this since it wasn't needed.